src/BugsnagUnity/Native/Android/NativeDevice.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace BugsnagUnity
+{
+    public class NativeDevice: NativePayloadClassWrapper, IDevice
+    {
+        public NativeDevice(AndroidJavaObject nativePointer) : base (nativePointer){}
+
+        public string BrowserName { get; set; }
+        public string BrowserVersion { get; set; }
+        public string ModelNumber { get; set; }
+        public string UserAgent { get; set; }
+
+
+        public string Id { get => GetNativeString("getId"); set => SetNativeString("setId", value); }
+        public string Locale { get => GetNativeString("getLocale"); set => SetNativeString("setLocale", value); }
+        public string Manufacturer { get => GetNativeString("getManufacturer"); set => SetNativeString("setManufacturer", value); }
+        public string Model { get => GetNativeString("getModel"); set => SetNativeString("setModel", value); }
+        public string OsName { get => GetNativeString("getOsName"); set => SetNativeString("setOsName", value); }
+        public string OsVersion { get => GetNativeString("getOsVersion"); set => SetNativeString("setOsVersion", value); }
+        public int? TotalMemory { get => GetNativeInt("getTotalMemory"); set => SetNativeLong("setTotalMemory", value); }
```

Comment:
I don't think this is guaranteed to fit in an int, and you might get weird behavior when you try to get or set a large number.

Reply:
This will be addressed in the next PR